doc/examples/viz_advanced.py

```diff
@@ -0,0 +1,187 @@
+
+"""
+Now we create, a ``Renderer`` object and add the streamlines using the ``line``
+function and an image plane using the ``slice`` function.
+"""
+
+ren = window.Renderer()
+stream_actor = actor.line(streamlines)
+
+if not world_coords:
+    image_actor = actor.slicer(data, affine=np.eye(4))
+else:
+    image_actor = actor.slicer(data, affine)
+
+"""
+For fun let's change also the opacity of the slicer
+"""
+
+slicer_opacity = .6
+image_actor.opacity(slicer_opacity)
+
+"""
+Connect the actors with the Renderer.
+"""
+
+ren.add(stream_actor)
+ren.add(image_actor)
+
+"""
+Now we would like to change the position of the ``image_actor`` using a slider.
+The sliders are widgets which require access to different areas of the
+visualization pipeline and therefore we don't recommend using them with
+``show``. The more appropriate way is to use them with the ``ShowManager``
+object which allows accessing the pipeline in different areas. Here is how:
+"""
+
+show_m = window.ShowManager(ren, size=(1200, 900))
+show_m.initialize()
+
+"""
+After we have initialized the ``ShowManager`` we can go ahead and create a
+callback which will be given to the ``slider`` function.
+"""
+
+
+def change_slice(obj, event):
+    z = int(np.round(obj.get_value()))
+    image_actor.display_extent(0, shape[0] - 1,
+                               0, shape[1] - 1, z, z)
+
+slider = widget.slider(show_m.iren, show_m.ren,
+                       callback=change_slice,
+                       min_value=0,
+                       max_value=shape[2] - 1,
+                       value=shape[2] / 2,
+                       label="Move slice",
+                       right_normalized_pos=(.98, 0.6),
+                       size=(120, 0), label_format="%0.lf",
+                       color=(1., 1., 1.),
+                       selected_color=(0.86, 0.33, 1.))
+
+"""
+Then, we can render all the widget and everything else in the screen and
+start the interaction using ``show_m.start()``.
+"""
+
+# show_m.render()
+# show_m.start()
+
+"""
+However, if you change the window size, the slider will not update its position
+properly. The solution to this issue is to update the position of the slider
+using its ``place`` method every time the window size changes.
+"""
+
+global size
+size = ren.GetSize()
+
+
+def win_callback(obj, event):
+    global size
+    if size != obj.GetSize():
+
+        slider.place(ren)
+        size = obj.GetSize()
+
+show_m.initialize()
+
+"""
+Finally, please uncomment the following lines so that you can interact with
+the available 3D and 2D objects.
+"""
+
+# show_m.add_window_callback(win_callback)
+# show_m.render()
+# show_m.start()
+
+ren.zoom(1.5)
+ren.reset_clipping_range()
+
+window.record(ren, out_path='bundles_and_a_slice.png', size=(1200, 900))
+
+"""
+.. figure:: bundles_and_a_slice.png
+   :align: center
+
+   **A few bundles with interactive slicing**.
+"""
+
+del show_m
```

Comment:
When you call this example from ipython with qt backend the window stays open because the show_m.initialize. So del makes sure it will go away. In python the del is not necessary.